frontend/src/folder/pages/EditFolder.tsx

```diff
@@ -0,0 +1,78 @@
+import { BsThreeDots } from 'react-icons/bs';
+import useFolderActions from '../hooks/useFolder';
+import { useFolderContext } from '../hooks/useFolderContext';
+import { Input } from '@/components/ui/input';
+import { Button } from '@/components/ui/button';
+
+export default function EditFolder() {
+  const {
+    newTitle,
+    setNewTitle,
+    isEditing,
+    handleEditClick,
+    showOptions,
+    handleCancelClick,
+    handleSaveClick,
+    toggleOptions,
+    optionsRef,
+    inputRef,
+    handleDeleteClick,
+  } = useFolderActions();
+  const { selectedFolderTitle } = useFolderContext();
+
+  return (
+    <div className="flex h-[44px] justify-between items-center border-b py-2 px-2">
+      {isEditing ? (
+        <Input
+          ref={inputRef}
+          value={newTitle}
+          onChange={(e) => setNewTitle(e.target.value)}
+          className="text-[18px] font-bold border-none focus:outline-none focus:border-none px-0 py-0"
+        />
+      ) : (
+        <h1 className="text-[18px] font-bold">{selectedFolderTitle}</h1>
+      )}
+
+      <div className="relative" ref={optionsRef}>
+        {isEditing ? (
+          newTitle.trim() === selectedFolderTitle ? (
+            <Button
+              onClick={handleCancelClick}
+              className="bg-transparent text-[14px] text-gray-500 font-normal hover:bg-transparent"
```

Comment:
You're passing a ton of classname to your button component almost everywhere you use it 

The points of having a design system is to avoid that — let your button component define all the variants that it needs and just use them without passing extra classnames to change it's styling

By default shadcn has the following variants for buttons:
```
const buttonVariants = cva(
  "inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50",
  {
    variants: {
      variant: {
        default: "bg-primary text-primary-foreground hover:bg-primary/90",
        destructive:
          "bg-destructive text-destructive-foreground hover:bg-destructive/90",
        outline:
          "border border-input bg-background hover:bg-accent hover:text-accent-foreground",
        secondary:
          "bg-secondary text-secondary-foreground hover:bg-secondary/80",
        ghost: "hover:bg-accent hover:text-accent-foreground",
        link: "text-primary underline-offset-4 hover:underline",
      },
      size: {
        default: "h-10 px-4 py-2",
        sm: "h-9 rounded-md px-3",
        lg: "h-11 rounded-md px-8",
        icon: "h-10 w-10",
      },
    },
    defaultVariants: {
      variant: "default",
      size: "default",
    },
  }
)
```

You can always change those properties and add new ones and instead of passing them as classnames you can just do `<Button variant="ghost" textColor="gray">` or sth

Reply:
Done!